### ISSUE TITLE
replace RosterID() with RosterInfo()

### ISSUE THREAD

**Issue by jasonalex13** [jasonalex13/CurRosterTool#55]
This is better naming for the future expansive use of this function. Also propose use of better layout of the enumerators for future expansion:

__CurRosterInfo(crew__*[, ReturnType]*__)__

Return types:
- `0` / __rosterenum__: enumerator of roster
- `1` / __crewenum__: enumerator of crew _(future use)_
- `10` / __generic__: generic name (either latest EBA name or assigned identifying)
- `11` / __descriptive__: descriptive name (like 7 Day 8.5 hour shift 5 panel)
- `12` / __alpha__: for letter crews, the letters (like FGHI, or ABC)
- `13` / __shifts__: which shift types that crew uses (eg: "D", "DN" or "DNA")
- `14` / __shiftlength__: default length of the shift _[number]_
- `15` / __coverage__: coverage of the roster (i.e. 5 days or 7 days) 
- `16` / __rostertype__: "Description" in the EBA (e.g. "Rotating Shift" or "Permanent Days") 
- `21` / __shiftsd__: whether the roster includes day shifts _[boolean]_ 
- `22` / __shiftsn__: whether the roster includes night shifts _[boolean]_ 
- `23` / __shiftsa__: whether the roster includes afternoonshifts _[boolean]_ 
- `30` / __EBA__: Current EBA name for the roster (for rosters defined in the current EBA, eg: Roster3)
- `31` / __EBAshort__: Shortform version of the current EBA name (eg: 3)
- `32` / __EA2012__: Name in 2012 EBA (for those defined in that document)
- `33` / __EA2015__: Name in 2015 EBA (for those defined in that document)

Blocks:
- __0-9:__ enumerators
- __10-19:__ general info
- __20-29:__ general info extension
- __30-39:__ EBA
- __40-49:__ _unused: reserved for future expansion of EBA_
- __50-59:__ _unused_
- __60-69:__ _unused_
- __70-79:__ _unused_
- __80-89:__ _unused_
- __90-99:__ _unused_

**Comment by jasonalex13** [comment]
the old __RosterID()__ function should point through to the new __RosterInfo()__ function for the remainder of this major version (3).

__javascript__
```javascript
function CurRosterID(crew, returntype) {
	if(typeof returntype != "number") {
		returntype = returntype.toUpperCase();
	}

	switch(returntype) {
		case 0:
		case "enum":
			returntype = 0;
			break;
		case 1:
		case "alpha":
			returntype = 12;
			break;
		case 2:
		case "EBA":
		case "EA":
			returntype = 30;
			break;
		case 3:
		case "EA2012":
		case "EBA2012":
			returntype = 32;
			break;
		case 4:
		case "EA2015":
		case "EBA2015":
			returntype = 33;
			break;
		case 5:
		case "EBAshort":
		case "EAshort":
			returntype = 31;
			break;
		default:
			returntype = 0;
	}
	output = CurRosterInfo(crew,returntype);
	return output;
}
```

__VBA__
```VB
Public Function CurRosterID(crew, Optional returntype = 0)
	Select Case returntype
	Case 0, "enum"
		returntype = 0
	Case 1, "alpha"
		returntype = 12
	Case 2, "EBA"
		returntype = 30
	Case 3, "EA2012"
		returntype = 32
	Case 4, "EA2015"
		returntype = 33
	Case 5, "EBAshort"
		returntype = 31
	Case Else
		returntype = 0
	End Select
End Function
```

It should be noted that the behavior in both languages above will return the __roster enum__ when a yet-undefined returntype is passed. It will need to be stressed in documentation that the correct returntypes should be used as this may not always return as expected.

**Comment by jasonalex13** [comment]
This feature has been implemented and will be merged through for v3.1
Marked as potential future in case the backwards compatibility code outlined here needs to be implemented later.
